packages/cloud/connection_details/kibana/global.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CoreStart } from '@kbn/core-lifecycle-browser';
+import type { CloudStart } from '@kbn/cloud-plugin/public';
+import type { SharePluginStart } from '@kbn/share-plugin/public';
+
+export interface ConnectionDetailsGlobalDependencies {
+  start: {
+    core: {
+      i18n: CoreStart['i18n'];
+      docLinks: CoreStart['docLinks'];
+      theme: CoreStart['theme'];
+      http: CoreStart['http'];
+      application: CoreStart['application'];
+      overlays: CoreStart['overlays'];
+    };
+    plugins: {
+      cloud?: CloudStart;
+      share?: SharePluginStart;
+    };
+  };
+}
+
+const kDependencies = 'ConnectionDetailsGlobalDependencies';
+
+export const setGlobalDependencies = (dependencies: ConnectionDetailsGlobalDependencies) => {
+  (global as any)[kDependencies] = dependencies;
```

Comment:
Coming to this PR with little context, it is not clear why we'd be choosing to pass dependencies on the global object like this. Is there a technical reason why we needed to do it this way?

AFAICT this only runs in the browser, don't we usually use react context providers for this type of dependency sharing?

Reply:
Vadim is on pto, I'll share a bit of context I have,

Here is the change of moving from explicit dependency passing making that makes that flyout easier to use: https://github.com/elastic/kibana/pull/180912/commits/ba59158511846fbaee9386d8ecde9319a5ba1b42


> AFAICT this only runs in the browser, don't we usually use react context providers for this type of dependency sharing?


I am not sure what is the best practice here. Couldn't find a good example. I assume it could be something like this? But I thought `useKibanaContext` is also something we don't encourage. 

```
function CardComponent() {
 const {start} = useKibanaContext() 

  // ...
 openConnectionDetails(start);
}
``` 

@clintandrewhall, does the usage of `openConnectionDetails` suppose to look something like this or could you share an example? 

> I remain concerned we've shipped attaching dependencies to globalThis.

This looks like the same thing we do with shared UI kbn bundle, just instead of making sure the code isn't copied, we're making sure we have a shared state.

Reply:
Hey Anton! Thanks for clarifying, that's really helpful background, and I hadn't seen the commit where the change was made.

I'll defer to Clint on the question about usage of the context provider.

But to the point about the shared UI kbn bundle, yes we do this, but it is handled behind the scenes at build time and nobody is accessing that object directly (it's transparent to users, they just import a package as they normally would so nobody ever needs to go mucking around directly with the global object). And anyway, the point of the shared bundle is that (in theory) everything there truly _is_ global and relevant to all Kibana plugins.

To me that's a very different situation from using the global object to circumvent how the plugin system is designed and store state specific to an individual component. I know wiring things up by passing dependencies is more work, but this is how stateful code is meant to be shared in the platform when you are crossing plugin/package boundaries. I don't think it is worth breaking out of that just to avoid ~100 lines of boilerplate. (However irritating that boilerplate might be).

Reply:
@Dosant I'm preparing a PR with a refactor that demonstrates the latest pattern.  Stay tuned, please!

Reply:
@clintandrewhall were you able to create the PR?

Reply:
@vadimkibana I started, but there are lot of implementation details that were unclear, and I've been pulled away by other concerns and illness.  I'll return to it next week.

In short:

- we need to refactor the global scope storage of dependencies;
- I'm wondering why there are so many unspecified options available;
- I'm curious why we have a class of observables in lieu of local component state (e.g. `showCloudId`)